mp/src/game/shared/momentum/weapon/weapon_mom_stickybomblauncher.cpp

```diff
@@ -0,0 +1,382 @@
+
+        StickybombHandle hHandle;
+        hHandle = pStickybomb;
+        m_Stickybombs.AddToTail(hHandle);
+
+        m_iStickybombCount = m_Stickybombs.Count();
+#endif
+    }
+    return pProjectile;
+}
+
+CBaseEntity *CMomentumStickybombLauncher::FireStickybomb(CMomentumPlayer *pPlayer)
+{
+#ifdef GAME_DLL
+
+    Vector vecForward, vecRight, vecUp;
+    AngleVectors(pPlayer->EyeAngles(), &vecForward, &vecRight, &vecUp);
+
+    // Create grenades here!!
+    Vector vecSrc = pPlayer->Weapon_ShootPosition();
+    vecSrc += vecForward * 16.0f + vecRight * 8.0f + vecUp * -6.0f;
+
+    Vector vecVelocity = (vecForward * 900.0f) + (vecUp * 200.0f) +
+                         (random->RandomFloat(-10.0f, 10.0f) * vecRight) + (random->RandomFloat(-10.0f, 10.0f) * vecUp);
+
+    CMomStickybomb *pProjectile = CMomStickybomb::Create(
+        vecSrc, pPlayer->EyeAngles(), vecVelocity, AngularImpulse(600, random->RandomInt(-1200, 1200), 0), pPlayer);
+
+    return pProjectile;
+#endif
+
+    return NULL;
+}
+
+void CMomentumStickybombLauncher::AddStickybomb(CMomStickybomb *pBomb)
+{
+    StickybombHandle hHandle;
+    hHandle = pBomb;
+    m_Stickybombs.AddToTail(hHandle);
+}
+
+//-----------------------------------------------------------------------------
+// Purpose: If a stickybomb has been removed, remove it from our list
+//-----------------------------------------------------------------------------
+void CMomentumStickybombLauncher::DeathNotice(CBaseEntity *pVictim)
+{
+    Assert(dynamic_cast<CMomStickybomb *>(pVictim));
+
+    StickybombHandle hHandle;
+    hHandle = (CMomStickybomb *)pVictim;
+    m_Stickybombs.FindAndRemove(hHandle);
+
+    m_iStickybombCount = m_Stickybombs.Count();
+}
+
+//-----------------------------------------------------------------------------
+// Purpose: Remove *with* explosions
+//-----------------------------------------------------------------------------
+bool CMomentumStickybombLauncher::DetonateRemoteStickybombs(bool bFizzle)
+{
+    bool bFailedToDetonate = false;
+
+    int count = m_Stickybombs.Count();
+
+    for (int i = 0; i < count; i++)
+    {
+        CMomStickybomb *pTemp = m_Stickybombs[i];
+        if (pTemp)
+        {
+            // This guy will die soon enough.
+            if (pTemp->IsEffectActive(EF_NODRAW))
+                continue;
+#ifdef GAME_DLL
+            if (bFizzle)
+            {
+                pTemp->Fizzle();
+            }
+#else
+            if (bFizzle == false)
+            {
+                if ((gpGlobals->curtime - pTemp->m_flCreationTime) < 0.8f) // Stickybomb arm time
+                {
+                    bFailedToDetonate = true;
+                    continue;
+                }
+            }
+#endif
+#ifdef GAME_DLL
+
+            pTemp->Detonate();
+#endif
+        }
+    }
+
+    return bFailedToDetonate;
+}
+
+float CMomentumStickybombLauncher::GetChargeMaxTime(void) { return MOM_STICKYBOMB_MAX_CHARGE_TIME; }
```

Comment:
You can remove `(void)` inside of parameters, it's an unnecessary idiom from the 90s